protostar/testing/environments/fuzz_test_execution_environment.py

```diff
@@ -37,6 +37,22 @@
 )
 
 
+def multiply_decorator(other_dec, args):
+    def inner(func):
+        current_fn = func
+        if not args:
+            return current_fn
+        args.reverse()
```

Comment:
[Tests] Why `args` reversed? Can you add a unit test for this decorator? Tests can be used as a form of documentation.

Reply:
The `args` should be reversed to keep the correct order of invoking. When you write your test setup like this:
```
example(1)
example(2)
example(3)
given(some strategy...)
```
you expect this test to be run with the following data in the following order:
```
test(1)
test(2)
test(3)
for x in data_from_strategy: test()
```

Now, the `multiply_decorator` wraps the current function with an additional `example` (basically it might be some other decorator, but in this case, we have the `example`) function with a specified set of arguments, so it would work like that (for the test I added in the beginning):
```
example(example(example(given(some strategy...), 1), 2), 3)
```
So you would get something like this
```
test(3)
test(2)
test(1)
for x in data_from_strategy: test()
```
That's why I put the `reverse` there.

> Can you add a unit test for this decorator?

Sure

Reply:
[add test for multiply decorator](https://github.com/software-mansion/protostar/pull/947/commits/361e96ec4a40e7e9b41092acfd7c4813ef13bb0b)